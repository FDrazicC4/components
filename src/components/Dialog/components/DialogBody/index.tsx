import React, { ReactElement } from 'react';
import clsx from 'clsx';
import Typography from '../../../Typography';
import { Props } from './types';


function DialogBody(props: Props): ReactElement {
    const { className, children, ...otherProps } = props;
    const classes = clsx(
        'brew-Dialog__body',
        className,
    );

    return (
        <Typography as="div" className={classes} {...otherProps}>{children}</Typography>
    );
}

export default DialogBody;
