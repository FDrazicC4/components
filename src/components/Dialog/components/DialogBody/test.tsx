import React from 'react';
import renderer from 'react-test-renderer';
import DialogBody from './index';
import { shallow } from 'enzyme';

describe('DialogBody', () => {
    it('matches the current snapshot', () => {
        const dialogBody = renderer.create(<DialogBody />).toJSON();
        expect(dialogBody).toMatchSnapshot();
    });

    it('mounts with the right classes', () => {
        expect(shallow(<DialogBody />).hasClass('brew-Dialog__body')).toBe(true);
    });

    it('passes through class names correctly', () => {
        const testClassName = `class-${Math.floor(Math.random() * 1000).toString()}`;
        expect(shallow(<DialogBody className={testClassName} />).hasClass(testClassName)).toBe(true);
    });

    it('accepts a children prop correctly', () => {
        const children = <span>children</span>;
        expect(shallow(<DialogBody>{children}</DialogBody>).contains(children)).toBe(true);
    });
});
