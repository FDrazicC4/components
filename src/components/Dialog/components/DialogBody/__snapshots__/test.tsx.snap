// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogBody matches the current snapshot 1`] = `
<div
  className="brew-Typography brew-Typography--variant-inherit brew-Typography--color-initial brew-Typography--align-inherit brew-Typography--display-initial brew-Dialog__body"
/>
`;
